InteractionBase/src/commands/CRemoveNodeFromView.cpp

```diff
@@ -37,14 +37,23 @@ CRemoveNodeFromView::CRemoveNodeFromView()
 bool CRemoveNodeFromView::canInterpret(Visualization::Item *source, Visualization::Item *target,
 	const QStringList &commandTokens, const std::unique_ptr<Visualization::Cursor> &cursor)
 {
-	 return CommandWithDefaultArguments::canInterpret(source, target, commandTokens, cursor)
-			 && target->hasNode() && target->scene()->currentViewItem()->allNodes().contains(target->node());
+	bool canInterpret = CommandWithDefaultArguments::canInterpret(source, target, commandTokens, cursor);
+	//The first parent with a node should be in the current view
+	while (source && !source->hasNode())
+		source = source->parent();
+	if (!source)
+		return false;
```

Comment:
For short statements, please put the then branch of the If (also the else, if it's short, and also short one line loop boides) on the same line.

Here, please put the return on the previous line.